airbyte-webapp/src/components/connectorBuilder/Builder/InputsView.tsx

```diff
@@ -222,21 +255,25 @@ const InputModal = ({
           />
           {values.type !== "unknown" ? (
             <>
-              <BuilderField
-                path="type"
-                type="enum"
-                options={["string", "number", "integer", "array", "boolean", "enum"]}
-                label={formatMessage({ id: "connectorBuilder.inputModal.type" })}
-                tooltip={formatMessage({ id: "connectorBuilder.inputModal.typeTooltip" })}
-              />
-              {values.type === "enum" && (
-                <BuilderField
-                  path="definition.enum"
-                  type="array"
-                  optional
-                  label={formatMessage({ id: "connectorBuilder.inputModal.enum" })}
-                  tooltip={formatMessage({ id: "connectorBuilder.inputModal.enumTooltip" })}
-                />
+              {!isInferredInputOverride && (
+                <>
+                  <BuilderField
+                    path="type"
+                    type="enum"
+                    options={["string", "number", "integer", "array", "boolean", "enum"]}
+                    label={formatMessage({ id: "connectorBuilder.inputModal.type" })}
+                    tooltip={formatMessage({ id: "connectorBuilder.inputModal.typeTooltip" })}
+                  />
+                  {values.type === "enum" && (
+                    <BuilderField
+                      path="definition.enum"
+                      type="array"
+                      optional
+                      label={formatMessage({ id: "connectorBuilder.inputModal.enum" })}
+                      tooltip={formatMessage({ id: "connectorBuilder.inputModal.enumTooltip" })}
+                    />
+                  )}
+                </>
               )}
               <BuilderField
                 path="definition.airbyte_secret"
```

Comment:
Should this be modifiable for inferred input overrides? It seems somewhat unsafe to let users make one of these fields non-secret, right?

Similar question for the default value - if we let users enter a default value, will that open them up to potentially saving credentials into their connector manifest?

I suppose the argument for doing these things this way are that they let users have more flexibility in these fields, in case they have a use case like always using the same username in their connector. So as long as we default to the most secure options (which we are doing), I think I'm okay with this.

Reply:
I went through the same thought process - however as you aren't 100% sure either let's go with minimal options for now. We can always allow more things later, but it's hard to take things away.

This is how it looks now:
<img width="574" alt="Screenshot 2022-12-21 at 12 01 05" src="https://user-images.githubusercontent.com/1508364/208890177-8ee75f24-3eb9-42dc-9220-88a196458696.png">